### PR TITLE
Implement permission handling

### PR DESCRIPTION
**Pull request by gitkrakel**
https://gitkraken.atlassian.net/browse/GKCS-5331

### CODE REVIEW

File: src/background.ts [hunk 1]
```diff
@@ -26,40 +33,97 @@ webNavigation.onHistoryStateUpdated.addListener(details => {
...
+	if (webNavigation.onDOMContentLoaded.hasListener(injectScript)) {
+		webNavigation.onDOMContentLoaded.removeListener(injectScript);
+	} else {
+		console.debug('Adding onDOMContentLoaded injection listener for the first time');
+	}
+	const allDomains = Object.values<string[]>(injectionDomains as any).flat();
+	webNavigation.onDOMContentLoaded.addListener(injectScript, {
+		url: allDomains.map((domain) => ({ hostContains: domain })),
+	});
+}
```
Comment (gitkrakel) [inline src/background.ts:74]: I guess this is all technically enterprise stuff so I can move this out too

**Comment by jdgarcia** [comment]
@gitkrakel On Chrome everyone works the same. On Firefox, It's not prompting me for permissions 
image

Installed by running `yarn package:firefox-no-sign` and using Firefox's `Load temporary add-on`

**Comment by jdgarcia** [comment]
I do see an icon with this tooltip after I close the popup, but clicking on the popup still doesn't show any prompt
image

File: src/domUtils.ts [hunk 2]
```diff
@@ -1,9 +1,12 @@
-export const createAnchor = (href: string, target?: string) => {
+export const createAnchor = (href: string, target?: string, callback?: () => void) => {
```
Comment (jdgarcia) [inline src/domUtils.ts:1]: (minor) just to keep the convention of naming the params the same as the attributes, can you call it `onclick`?

File: src/gkApi.ts [hunk 3]
```diff
@@ -7,13 +7,22 @@ const gkDotDevUrl = 'https://gitkraken.dev';
...
+		return cookie?.value;
+	} catch (e) {
+		if ((e as Error)?.message.includes('No host permissions for cookies at url')) {
+			// ignore as we are waiting for required permissions
+		} else {
+			// otherwise log error and continue as if logged out
+			console.error(e);
+		}
+	}
+	return undefined;
```
Comment (jdgarcia) [inline src/gkApi.ts:25]: instead of try/catch, would it be better to check to see if we have the permission first?

Comment (gitkrakel) [reply src/gkApi.ts:25]: yeah i felt bad cause it adds an extra await but it's cleaner. will do :+1:

**Review by jdgarcia — APPROVED** [review]

**Pull request merged**